trulens_eval/trulens_eval/database/sqlalchemy.py

```diff
@@ -711,9 +736,33 @@ def __init__(self):
         self.feedback_columns = set()
 
     def get_df_and_cols(
-        self, apps: Iterable[orm.AppDefinition]
+        self,
+        apps: Optional[List[orm.AppDefinition]] = None,
+        records: Optional[List[orm.Record]] = None
     ) -> Tuple[pd.DataFrame, Sequence[str]]:
-        df = pd.concat(self.extract_apps(apps))
+        """Produces a records dataframe which joins in information from apps and
+        feedback results.
+
+        Args:
+            apps: If given, includes all records of all of the apps in this
+                iterable.
+
+            records: If given, includes only these records. Mutually exclusive
+                with `apps`.
+        """
+
+        assert apps is None or records is None, "`apps` and `records` are mutually exclusive"
+
+        if apps is not None:
+            df = pd.concat(self.extract_apps(apps))
+
+        elif records is not None:
+            apps = set(record.app for record in records)
```

Comment:
Paginating in this way may not reveal all apps if some AppDefinition is out of range of the limit & offset. If so we will have to change how the evaluations page finds App IDs to select from since that's currently given from the `df` here

Reply:
Changed eval page to get the apps independently of this changed method.